quinn-proto/src/connection/pacing.rs

```diff
@@ -0,0 +1,73 @@
+//! Pacing of packet transmissions.
+
+use std::{
+    convert::TryInto,
+    time::{Duration, Instant},
+};
+
+/// A simple token-bucket pacer. The bucket starts full and has an adjustable capacity. Once the
+/// bucket is empty, further transmission is blocked. The bucket refills at a rate determined by
+/// the congestion window and estimated round-trip time.
+pub struct Pacer {
+    capacity: u64,
+    tokens: u64,
+    last_instant: Instant,
+}
+
+impl Pacer {
+    /// Obtains a new [`Pacer`].
+    pub fn new(capacity: u64) -> Self {
+        Self {
+            capacity,
+            tokens: capacity,
+            last_instant: Instant::now(),
+        }
+    }
+
+    /// Return how long we need to wait before sending a packet.
+    ///
+    /// If we can send a packet right away, this returns [`None`]. Otherwise, returns `Some(d)`,
+    /// where `d` is the time before this function should be called again.
+    ///
+    /// We panic if the passed-in instant is older than `self.last_instant`.
+    pub fn delay(
+        &mut self,
+        smoothed_rtt: Duration,
+        mtu: u16,
+        window: u64,
+        mut now: Instant,
+    ) -> Option<Instant> {
+        if let Some(time_elapsed) = now.checked_duration_since(self.last_instant) {
+            self.last_instant = now;
+            let new_tokens = window.saturating_mul(5) / 4;
```

Comment:
The 5 / 4 ratio used twice here could use some comments/documentation on why/how that's chosen.

Reply:
It comes from the current base draft as of this writing, which [suggests N = 1.25](https://github.com/quicwg/base-drafts/blob/5e3723af74412f87907ec50025bd9c7510e8ad27/draft-ietf-quic-recovery.md).